staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go

```diff
@@ -183,6 +231,9 @@ func setNestedFieldNoCopy(obj map[string]interface{}, value interface{}, fields
 			if valMap, ok := val.(map[string]interface{}); ok {
 				m = valMap
 			} else {
+				if invalidTypeDetectionEnabled {
```

Comment:
do we do deep type checking? We should, go recursively go into the `map[string]interface{}`, compare the deepcopy func.